Test/CoreSDK.Test/Shared/Extensibility/MetricAggregatorManagerTest.cs

```diff
@@ -0,0 +1,606 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility
+{
+    using System;
+    using System.Reflection;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using System.Collections.ObjectModel;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.TestFramework;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Assert = Xunit.Assert;
+
+    [TestClass]
+    public class MetricAggregatorManagerTest
+    {
+        [TestMethod]
+        public void MetricAggregatorMayBeCreatedForMetricHavingNoDimensions()
```

Comment:
I felt these integration tests were hard to understand for someone who is not already familiar with all 3 classes implementing metric aggregation. Arrange, Act, Assert comments would help. If it's possible at this point, factoring the classes to allow unit testing would help more.